lib/event_tracer/buffer.rb

```diff
@@ -0,0 +1,65 @@
+require 'concurrent'
+
+module EventTracer
+  # This is an implementation of buffer storage. We use Concurrent::Array underneath
+  # to ensure thread-safe behavior. Data is stored until certain size / interval
+  # before flushing.
+  #
+  # Caveats: We should only store non-important data like logs in this buffer
+  # because if a process is killed, the data in this buffer is lost.
+  class Buffer
+    # Buffer can store maximum 10 items.
+    # Bigger size requires more memory to store, so choose a reasonable number
+    DEFAULT_BUFFER_SIZE = 10
+    # An item can live in buffer for at most 10s between each `Buffer#add`
```

Comment:
Sorry, this is my mistake, but it should be `at least`
```suggestion
    # An item can live in buffer for at least 10s between each `Buffer#add`
```